### ISSUE TITLE
Add a Settings Screen to handle user preferences.

### ISSUE THREAD

**Issue by TheWebTech** [williamspiro/HubSpot-Developer-Extension#8]
For certain features it makes sense to store user preferences. 
For example @GonzaloTorreras and I want to add a togglable dark theme option which if turned on automatically applies a stylesheet when the design manager is open.
That's a preference - not everyone is going to want that, plus if HS makes a significant change one day it'd be good to have a way to disable it.

Question is how do we want to handle the options screen.

Do we make it a right click "Options" item in the context menu? that seems to be a common thing. Or do we make it exist inside the dropdown where you click a button and it reveals the options screen?

I'm leaning toward options in context menu as that just seems to be the common thing to do.

**Comment by GonzaloTorreras** [comment]
I like the idea to have everything accesible through the popup more than a option page, but the true is that if this start to become bigger and more customizable we will need to move to an option page... so don't have the answer.

I will add a new branch so you can see all I had:
- option page with one setting that is saved through all your sessions.
- CSS of the dark theme (need to be cleaned, organiced and fix some elements that aren't well styled yet).
- The injector of the CSS (always active).

So the next logic step is integrate both and set the conditional to load the CSS only in the option is active.

**Comment by TheWebTech** [comment]
Well how about this then - Personally I think it'd visually look cool to have the options show in the popup until we have more settings to actually configure.

How about we just consciously build it in such a way that the CSS HTML and any required JS will be easy to just split off later on?

**Comment by GonzaloTorreras** [comment]
Looks good for me.
Actually thinking on that, probably there will be some fast-settings that we will want always available and others than can be like user prefferences on the option page. For example:
Popup will handle the switch on/off for dark theme.
But in the option page will be a selector with the dark theme you want to use.

So following this logic we can have a good approach about where should be handle the options.

Temporary I see fine use just the popup 👍

**Comment by TheWebTech** [comment]
makes sense. Always have a quick options area, and later add a more detailed options page when we have more choices to make. We'll just move the individual toggles and controls over to the options page as needed to make room for things in the quick settings as we go.

**Comment by TheWebTech** [comment]
Going to make just a few updates and will release the settings page as it is basically to get those two features live. 

We'll implement https://github.com/williamspiro/HubSpot-Developer-Extension/issues/32
in combo with the ideas talked about here.

Just figure no point in holding off releasing two features due to a mostly organizational change.
